### PR TITLE
Scanning scsi bus removed as it sometimes stuck in the middle

### PR DESCRIPTION
**Pull request by Madhan-SWE**
Fixes: https://github.com/ppc64le-cloud/powervs-csi-driver/issues/19

### CODE REVIEW

**Comment by mkumatag** [comment]
This code is derived from here: https://github.com/kubernetes-csi/csi-lib-fc, maybe we should start using that module and avoid making any changes here.

**Comment by Madhan-SWE** [comment]
> This code is derived from here: https://github.com/kubernetes-csi/csi-lib-fc, maybe we should start using that module and avoid making any changes here.

Yes Correct.
I will make necessary changes.

File: pkg/util/util.go [hunk 1]
```diff
@@ -92,3 +94,13 @@ func GetAccessModes(caps []*csi.VolumeCapability) *[]string {
 	}
 	return &modes
 }
+
+func RemoveMultipathDevice(device string) error {
```
Comment (mkumatag) [inline pkg/util/util.go:98]: We should aim to push this code to https://github.com/kubernetes-csi/csi-lib-fc

**Review by mkumatag — CHANGES_REQUESTED** [review]
make sure to compare with the upstream code and see whats the difference is in this file?! and also run good amount of tests to confirm this is working fine.

File: pkg/util/util.go (hunk 1 above)
Comment (Madhan-SWE) [reply pkg/util/util.go:98]: PR Raised: https://github.com/kubernetes-csi/csi-lib-fc/pull/4

**Review by mkumatag — APPROVED** [review]
This can be merged now, later whenever we merge the upstream code then we can replace with that.

/lgtm
/hold

added hold as well, remove the hold whenever you are comfortable.

**Comment by mkumatag** [comment]
@Madhan-SWE I see conflicts here as well, the following instructions can help you rebase your code.

```
git remote add upstream https://github.com/ppc64le-cloud/powervs-csi-driver.git
git fetch upstream
git checkout main
git merge upstream/main
git checkout host_scan_optimization
git merge origin/main
```

and also fix the build error

**Comment by Madhan-SWE** [comment]
Raised separate issue to use upstream fibre channel.
This PR can be released and merged.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**